Src/Middlewares/UserMiddleware.ts

```diff
@@ -0,0 +1,30 @@
+import { Request, Response, NextFunction } from "express";
+
+export const validateUser = (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	const { firstName, lastName, email, password } = req.body;
+	const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+	const passwordRegex = /^(?=.*\d)(?=.*[a-zA-Z])[a-zA-Z\d]{8,}$/;
```

Comment:
I could never get past this password regex. To me, this regex is a bit too complex for what you are trying to achieve. min of 8 characters including special characters should not be this long.

To validate your regular expressions, try this tool https://regexr.com/